optaplanner-core/src/main/java/org/optaplanner/core/impl/io/OptaPlannerXmlSerializationException.java

```diff
@@ -16,21 +16,21 @@
 
 package org.optaplanner.core.impl.io;
 
-public class XmlUnmarshallingException extends RuntimeException {
+public class OptaPlannerXmlSerializationException extends RuntimeException {
```

Comment:
open question (maybe for another PR): Is this public API? Should SolverFactory.createFromXmlResource()'s javadoc have a "@throws OptaPlannerXmlSerializationException" ? In that case we might want to move it into the api package.

Reply:
Naming:
- We avoid class names that contain the word "OptaPlanner" because everything is OptaPlanner.
- Is it Serialization or Marshalling? Much of the other code naming and error messages seem to talk about marshalling, not serialization. Let's use one term and stick with that.

Reply:
The same question applies to all the fail-fast checks: should we mention there the `IllegalArgumentException`? Maybe it should. In that case, I would definitely do it in a separate PR.